File: resources/Torus World/scripts/RockProperties.js
RockProperties = Class({
    Extends : CostumePropertiesScript,

    init : function() {
        this.pieces = 0;
        this.points = 10;
        this.strength = 0;
        this.hitsRequired = 1;
    },

    getProperties : function() {
        return RockProperties.properties;
    }
});

RockProperties.properties = new java.util.ArrayList();
RockProperties.properties.add(new itchy.property.IntegerProperty("pieces"));
RockProperties.properties.add(new itchy.property.IntegerProperty("points"));
RockProperties.properties.add(new itchy.property.IntegerProperty("strength"));
RockProperties.properties.add(new itchy.property.IntegerProperty("hitsRequired"));

File: resources/Space-Miner/scripts/rockProperties.py
from common import *

properties = ArrayList()
properties.add( IntegerProperty("pieces") )
properties.add( IntegerProperty("points") )
properties.add( IntegerProperty("strength") )
properties.add( IntegerProperty("hitsRequired") )
 
class RockProperties(CostumeProperties) :

    def __init__(self) :
    	self.pieces = 0
    	self.points = 10
    	self.strength = 0
    	self.hitsRequired = 1


    # Boiler plate code - no need to change this
    def getProperties(self):
        return properties

    # Boiler plate code - no need to change this
    def getClassName(self):
        return ClassName( CostumeProperties, self.__module__ + ".py" )

